app/src/androidTest/java/com/github/sdp_begreen/begreen/viewModels/ConnectedUserViewModelTest.kt

```diff
@@ -0,0 +1,246 @@
+package com.github.sdp_begreen.begreen.viewModels
+
+import android.graphics.Bitmap
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.MediumTest
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
+import com.github.sdp_begreen.begreen.models.PhotoMetadata
+import com.github.sdp_begreen.begreen.models.User
+import com.github.sdp_begreen.begreen.rules.CoroutineTestRule
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.test.*
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.contains
+import org.junit.Assert.assertThrows
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.KoinTestRule
+import org.mockito.Mockito.*
+
+/**
+ * I started by putting this test in the unit test package, but I needed support from android
```

Comment:
in general avoid `I` comments, the code is owned by the team

Reply:
Isn't it technically owned by EPFL since it's part of a course? :laughing:

Reply:
You are right. I used the `I` comments here to let know my team why I had to test this `viewModel` in `androitTest` and not in normal `unitTest`. But you are right that it's not the good way to do it. I will change this comment.